services/director-v2/src/simcore_service_director_v2/models/domains/comp_tasks.py

```diff
@@ -24,16 +24,31 @@
 class Image(BaseModel):
     name: str = Field(..., regex=KEY_RE)
     tag: str = Field(..., regex=VERSION_RE)
-    requires_gpu: Optional[bool] = Field(None, deprecated=True)
-    requires_mpi: Optional[bool] = Field(None, deprecated=True)
+
+    requires_gpu: Optional[bool] = Field(
+        None, deprecated=True, description="Use instead node_requirements"
+    )
+    requires_mpi: Optional[bool] = Field(
+        None, deprecated=True, description="Use instead node_requirements"
+    )
     node_requirements: NodeRequirements = Field(
         None, description="the requirements for the service to run on a node"
     )
+    command: list[str] = Field(
+        default=[
+            "run",
+        ],
+        description="command to run container. Can override using ContainerSpec service labels",
+    )
 
     @validator("node_requirements", pre=True, always=True)
     @classmethod
     def migrate_from_requirements(cls, v, values):
         if v is None:
+            # NOTE: 'node_requirements' field's default=None although is NOT declared as nullable.
+            # Then this validator with `pre=True, always=True` is used to create a default
+            # based on that accounts for an old version.
+            # This strategy guarantees backwards compatibility
```

Comment:
ok I see your point. I guess we'll iterate over this again with the personalized resources case again.

Reply:
yes.